frontend/src/components/mine/Permit/MinePermitInfo.js

```diff
@@ -107,49 +99,54 @@ const getPermittees = (partyRelationships, permits) =>
+const transformRowData = (permit, partyRelationships) => {
+  const latestAmendment = permit.amendments[0];
+  const firstAmendment = permit.amendments[permit.amendments.length - 1];
 
-  const permittees = getPermittees(partyRelationships, permits);
+  const permittees = getPermittees(partyRelationships, permit);
   const permitteeName =
     partyRelationships.length === 0 ? Strings.LOADING : getPermitteeName(permittees);
 
   return {
-    key: latest.permit_guid,
-    lastAmended: formatDate(latest.issue_date),
-    permitNo: latest.permit_no || Strings.EMPTY_FIELD,
-    firstIssued: formatDate(first.issue_date) || Strings.EMPTY_FIELD,
+    key: permit.permit_guid,
+    lastAmended: (latestAmendment && formatDate(latestAmendment.issue_date)) || Strings.EMPTY_FIELD,
+    permitNo: permit.permit_no || Strings.EMPTY_FIELD,
+    firstIssued: (firstAmendment && formatDate(firstAmendment.issue_date)) || Strings.EMPTY_FIELD,
     permittee: permitteeName,
-    authorizationEndDate: latest.authorization_end_date
-      ? formatDate(latest.authorization_end_date)
-      : Strings.EMPTY_FIELD,
-    permittees,
-    status: Strings.EMPTY_FIELD,
+    authorizationEndDate:
+      latestAmendment && latestAmendment.authorization_end_date
+        ? formatDate(latestAmendment.authorization_end_date)
+        : Strings.EMPTY_FIELD,
+    amendments: permit.amendments,
+    status: permit.permit_status_code,
   };
 };
 
-const transformChildRowData = (permittee, record) => ({
-  key: record.key,
-  permitNo: record.permitNo,
-  Date: permittee.start_date,
-  permittee: permittee.party.name,
+const transformChildRowData = (amendment, record, amendmentNumber) => ({
+  key: amendmentNumber,
+  ReceivedDate: amendment.received_date,
+  IssueDate: amendment.issue_date,
+  AuthorizationEndDate: amendment.authorization_end_date,
   description: Strings.EMPTY_FIELD,
 });
 
 export const MinePermitInfo = (props) => {
-  const groupedPermits = Object.values(groupPermits(props.mine.mine_permit));
   const amendmentHistory = (record) => {
-    const childRowData = record.permittees.map((permittee) =>
-      transformChildRowData(permittee, record)
+    const childRowData = record.amendments.map((amendment, index) =>
+      transformChildRowData(amendment, record, record.amendments.length - index)
     );
     return (
       <Table align="left" pagination={false} columns={childColumns} dataSource={childRowData} />
     );
   };
-  const rowData = groupedPermits.map((permits) =>
-    transformRowData(permits, props.partyRelationships)
-  );
+  const rowData = props.mine.mine_permit
+    .filter((permit) => permit.permit_no.toUpperCase().charAt(1) !== "X")
+    .map((permit) => transformRowData(permit, props.partyRelationships))
+    .concat(
+      props.mine.mine_permit
+        .filter((permit) => permit.permit_no.toUpperCase().charAt(1) === "X")
+        .map((permit) => transformRowData(permit, props.partyRelationships))
+    );
```

Comment:
I'm sure there's a good reason for this, but at a quick look, it seems like two mutually-exclusive parts of the `mine_permit` list are being transformed in the same way and then being joined.

What's the reason for splitting them apart before passing them to `transformRowData` and then joining them again? Any chance this reason could be made a bit more self-documenting in the code?

Reply:
I figured I would get called out on this, but it seemed like the quickest way to achieve the separation between exploration and other types of permits. I'm not confidant that we have all the requirements for "permit sorting", so I thought going this approach for now wouldn't require changing any of the other logic on this page.

Might also actually make some sense for the back-end to do this, if we want this to be the order universally (like on the mine search results). If I can confirm that with Aaron, I'll make that change and refactor this out in our current sprint.

Reply:
Ended up moving this to the API and removing this logic from the front-end entirely.